plugin/action/throttle/throttle.go

```diff
@@ -108,25 +142,57 @@ func (p *Plugin) Start(config pipeline.AnyConfig, params *pipeline.ActionPluginP
 	p.limiterBuff = make([]byte, 0)
 
 	limitersMu.Lock()
-	limiters[p.pipeline] = map[string]*limiter{}
+	limiters[p.pipeline] = map[string]limiter{}
 	limitersMu.Unlock()
 
+	if p.config.LimiterKind == redisLimiterType {
+		p.redisClient = redis.NewClient(&redis.Options{
+			Network:  "tcp",
+			Addr:     p.config.RedisHost,
+			Password: p.config.RedisPassword,
+		})
+
+		pingResp := p.redisClient.Ping()
+		if pingResp.Err() != nil {
+			logger.Fatalf("Can't ping redis: %s", pingResp.Err())
+		}
+	}
+
 	for _, r := range p.config.Rules {
 		p.rules = append(p.rules, NewRule(r.Conditions, complexLimit{r.Limit, r.LimitKind}))
 	}
 
 	p.rules = append(p.rules, NewRule(map[string]string{}, complexLimit{p.config.DefaultLimit, p.config.LimitKind}))
 }
 
+func (p *Plugin) getNewLimiter(throttleField, limiterKey string, rule *rule) limiter {
+	switch p.config.LimiterKind {
+	case redisLimiterType:
+		return NewRedisLimiter(
+			p.redisClient,
+			throttleField,
+			limiterKey,
+			p.config.RedisRefreshInterval_,
+			p.config.BucketInterval_,
+			p.config.BucketsCount,
+			rule.limit,
+			p.config.IsThrottleDistributed,
+		)
+	case inMemoryLimiterType:
```

Comment:
Can we skip this branch?

Reply:
It seems more readable.

Reply:
This code was changed and `fallthrough` were removed